layers/+lang/crystal/packages.el

```diff
@@ -34,7 +34,7 @@
     play-crystal))
 
 (defun crystal/post-init-company ()
-  (spacemacs//crystal-setup-company))
+  (add-hook 'crystal-mode-local-vars-hook #'spacemacs//crystal-setup-company))
```

Comment:
Here we are possibly setting the company backends multiple times. Is there a side effect with company when we register a backend multiple times and what happens when we set two different backends for the same mode.

With this directory-local-variable support we seem to have to call some functions multiple times, the same may apply to prefixes and key bindings. However if we really have someone wanting to switch between lsp and non-lsp backends we will most probably not have a clean key map anymore i.e. entering a python project with anaconda, then switching to another one with lsp the prefixes from anaconda won't be removed possibly hiding prefixes from `lsp` layer.

I am not sure how to handle this @syl20bnr @lebensterben.